extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourcesBuildTimeConfig.java

```diff
@@ -55,22 +58,17 @@ public class DataSourcesBuildTimeConfig {
      * @deprecated
      */
     @Deprecated
-    public Optional<String> url;
+    Optional<String> url();
 
     /**
      * Only here to detect configuration errors.
      *
      * @deprecated
      */
     @Deprecated
-    public Optional<String> driver;
+    Optional<String> driver();
 
-    public DataSourceBuildTimeConfig getDataSourceRuntimeConfig(String dataSourceName) {
-        if (DataSourceUtil.isDefault(dataSourceName)) {
-            return defaultDataSource;
-        }
-        DataSourceBuildTimeConfig dataSourceBuildTimeConfig = namedDataSources.get(dataSourceName);
-        return Objects.requireNonNullElseGet(dataSourceBuildTimeConfig, DataSourceBuildTimeConfig::new);
+    default boolean hasNamedDataSources() {
+        return dataSources().keySet().size() > 1 || !dataSources().containsKey(DataSourceUtil.DEFAULT_DATASOURCE_NAME);
     }
```

Comment:
_Ackshually_... this would return true for an empty config. Which might be fine, but I wouldn't take the risk?

```suggestion
    default boolean hasNamedDataSources() {
        return dataSources().keySet().size() > 1
                        || !dataSources().isEmpty()
                        && !dataSources().containsKey(DataSourceUtil.DEFAULT_DATASOURCE_NAME);
    }
```